Feature/doc glossary

m3d opened the pull request:
Added draft of glossary - ignore howto.rst, to be rebased or merged after #101

m3d commented:
Generated result is available at https://robotika.github.io/osgar/index.html

zwn commented on doc/sphinx/glossary.rst:
```diff
@@ -0,0 +1,36 @@
+      Binary file containing message data with their source and timestamp.
+
+   node
+      The basic unit attached to `bus`. It has fixed set of inputs and outputs.
```
I think I don't like having both `node` and `driver`. Can we have just `node`?

m3d replied:
Node is "everything" while "driver" is very specific to HW ... somehow I still see value using driver.

zwn replied:
I don't say it often but when I do, I do mean it.

m3d replied:
Sorry, I already overlooked/forgot this bit :(.

zwn commented on doc/sphinx/glossary.rst:
```diff
@@ -0,0 +1,36 @@
+      to a machine with set of sensors and efectors in the real world.
+
+   stream
+      Output from node without specified destination like for `link`.    
```
That's interesting. I have not realized that we can say that a `stream` is one output of a `node`. I actually had to go to the code to check it. Hmm.

m3d replied:
I use `stream` in logger.py when you actually define output pin of some module/node. This is different from `link` which connects two nodes ... what do you think?

zwn commented:
This page http://www.sphinx-doc.org/en/master/usage/quickstart.html says

> This markup will create a link to the documentation of enumerate().
> ```
> The :py:func:`enumerate` function can be used for ...
> ```

zwn commented:
I think we might want [autodoc](http://www.sphinx-doc.org/en/master/usage/quickstart.html#autodoc).

m3d commented:
Yes, I would use autodoc later on - again I am not sure if with this PR? I could also rebase it on master and ignore "dependency" on howto.rst ... thanks m.

m3d commented:
well, the missing part for apidoc was
```
m:\git\osgar\doc\sphinx>sphinx-apidoc -o . ..\..\osgar
Creating file .\osgar.rst.
Creating file .\osgar.drivers.rst.
Creating file .\osgar.lib.rst.
Creating file .\modules.rst.
```
based on https://stackoverflow.com/questions/13838368/no-generation-of-the-module-index-modindex-when-using-sphinx

but I am not sure if we want to keep them with the code??
See https://robotika.github.io/osgar/py-modindex.html

m3d commented:
I added also the generated *.rst files so that you know what I was talking about ...

zwn commented on doc/sphinx/msgtypes.rst:
```diff
@@ -0,0 +1,31 @@
+   pose2d
+      [x, y, heading] - position in meters with heading in radians
+      (anticlockwise). Scale 1000x (i.e. x, y are send in millimeters)
+      and angle in 1/100th of degree.
```
I don't understand "heading in radians" vs "angle in 1/100th of degree" while the transformation should be "Scale 1000x".

m3d replied:
I removed the scaling factor as "implementation detail" which we want to hide anyway.

zwn replied:
You removed it from the description but the actual messages used in the code are still integers with the scaling, right? So what is the plan with that? I am confused now. I propose that the documentation follows what is implemented in contrary to what might be implemented in the future.

```suggestion
pose2d
      [x, y, heading] - position in millimeters with heading in 1/100th of degree.
```

m3d replied:
OK, now there is a note about scaling factor, thanks.

zwn commented:
> I am not sure if we want to keep them with the code??

What is the planed workflow? If the files are going to be always regenerated, I see no point in keeping them. If there are any hand edits to make them work, it might be worth it.

m3d commented:
One option is to manually generate documentation website with a release of new package (say once a month?). Or have some "latest" version corresponding to unreleased master. Having `osgar.rst` in repository could help to generate changelog, if we will do it with releases. What is your preference? I guess that it is possible to have multiple doc versions on http://robotika.github.io/osgar/?
Maybe we should have version in URL like http://robotika.github.io/osgar/0.0.1/ ??

zwn commented:
I like the version in the url. I really like links that don't break 😄.

Having a changelog seems like a good idea.

m3d commented:
> Having a changelog seems like a good idea.

Does this mean, that you would keep `osgar.rst` in repository?

zwn commented:
> > Having a changelog seems like a good idea.
> 
> Does this mean, that you would keep `osgar.rst` in repository?

I don't know. Does it somehow help us to get the changelog? I don't see how.

m3d commented:
I rebased this PR on master and dropped commit with autogenerated RST files. There is a new call in `make.bat`. I would propose to merge it independently of `howto.rst`. I agree that QT glossary could be good reference, but maybe with another release ... OK?

m3d commented:
Let's merge it and revise it on demand (in particular I would like to use/see code documentation). I expect another iteration after osgar 0.1.0 release.
